libcni/api.go

```diff
@@ -193,24 +207,77 @@ func (c *CNIConfig) getResultCacheFilePath(netName string, rt *RuntimeConf) stri
 	return filepath.Join(c.getCacheDir(rt), "results", fmt.Sprintf("%s-%s-%s", netName, rt.ContainerID, rt.IfName))
 }
 
-func (c *CNIConfig) setCachedResult(result types.Result, netName string, rt *RuntimeConf) error {
+func (c *CNIConfig) setCachedResult(result types.Result, config []byte, netName string, rt *RuntimeConf) error {
+
+	cached := cachedInfo{Config: config,
+		Kind:       "cniCacheV1",
+		CNIVersion: "0.4.0"}
+	if len(rt.Args) > 0 {
+		cached.CniArgs = utils.StringifyArgs(rt.Args)
+	}
+	if len(rt.CapabilityArgs) > 0 {
+		cached.CapabilityArgs = rt.CapabilityArgs
+	}
+
 	data, err := json.Marshal(result)
 	if err != nil {
 		return err
 	}
+
 	fname := c.getResultCacheFilePath(netName, rt)
```

Comment:
@mccv1r0 Can we move this block (getResultCacheFilePath and MkdirAll() to just before the WriteFile()? And then also add a comment about the dance with RawResult/Result and the marshalling.